pkg/plugins/framework/helpers.go

```diff
@@ -0,0 +1,174 @@
+	}
+	return &config, nil
+}
+
+// CreateResponse creates a success response
+func CreateResponse(success bool, message string) *plugins.Response {
+	return &plugins.Response{
+		Success: success,
+		Message: message,
+	}
+}
+
+// CreateErrorResponse creates an error response
+func CreateErrorResponse(err error) *plugins.Response {
+	return &plugins.Response{
+		Success: false,
+		Message: err.Error(),
+	}
+}
+
+// EnsureDir ensures a directory exists, creating it if necessary
+func EnsureDir(path string) error {
+	if err := os.MkdirAll(path, 0755); err != nil {
+		return fmt.Errorf("failed to create directory %s: %w", path, err)
+	}
+	return nil
+}
+
+// GetInstanceDir returns the instance directory path
+func GetInstanceDir(instanceID string) string {
+	return filepath.Join(os.Getenv("HOME"), ".lima", instanceID)
+}
+
+// GetInstanceConfigPath returns the instance config file path
+func GetInstanceConfigPath(instanceID string) string {
+	return filepath.Join(GetInstanceDir(instanceID), "lima.yaml")
+}
+
+// GetInstanceDiskPath returns the instance disk file path
+func GetInstanceDiskPath(instanceID string) string {
+	return filepath.Join(GetInstanceDir(instanceID), "disk.img")
+}
+
+// GetInstanceSocketPath returns the instance socket file path
+func GetInstanceSocketPath(instanceID string) string {
+	return filepath.Join(GetInstanceDir(instanceID), "socket")
+}
+
+// GetPluginSocketPath returns the plugin socket file path
+func GetPluginSocketPath(pluginName string) string {
+	return filepath.Join("/tmp", fmt.Sprintf("lima-plugin-%s.sock", pluginName))
+}
+
+// CreateDiskImage creates a disk image with the given size
+func CreateDiskImage(path string, sizeGB int) error {
+	cmd := exec.Command("qemu-img", "create", "-f", "qcow2", path, fmt.Sprintf("%dG", sizeGB))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to create disk image: %s: %w", string(output), err)
+	}
+	return nil
+}
+
+// ResizeDiskImage resizes a disk image to the given size
+func ResizeDiskImage(path string, sizeGB int) error {
+	cmd := exec.Command("qemu-img", "resize", path, fmt.Sprintf("%dG", sizeGB))
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to resize disk image: %s: %w", string(output), err)
+	}
+	return nil
+}
+
+// CreateSnapshot creates a snapshot of a disk image
+func CreateSnapshot(path, snapshotPath string) error {
+	cmd := exec.Command("qemu-img", "create", "-f", "qcow2", "-b", path, snapshotPath)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to create snapshot: %s: %w", string(output), err)
+	}
+	return nil
+}
+
+// DeleteSnapshot deletes a snapshot
+func DeleteSnapshot(snapshotPath string) error {
+	if err := os.Remove(snapshotPath); err != nil {
+		return fmt.Errorf("failed to delete snapshot: %w", err)
+	}
+	return nil
+}
+
+// WaitForSocket waits for a Unix socket to become available
+func WaitForSocket(path string, timeout time.Duration) error {
```

Comment:
You can just use stdio instead of sockets
https://github.com/moby/buildkit/blob/de56a3c5056341667b5bad71f414ece70b50724f/frontend/gateway/grpcclient/client.go#L1240